new-components/scanners/gosec/internal/transformer/parser.go

```diff
@@ -0,0 +1,79 @@
+package transformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/owenrumney/go-sarif/v2/sarif"
+)
+
+type (
+	// parser implements sarif parsing for gosec out.
+	parser struct {
+		rawOutFilePath string
+		rawOut         string
+	}
+)
+
+// NewParser returns a new parser.
+func NewParser(rawOutFilePath string) (*parser, error) {
+	if rawOutFilePath == "" {
+		return nil, fmt.Errorf("invalid raw output file path: '%s'", rawOutFilePath)
+	}
+
+	return &parser{
+		rawOutFilePath: rawOutFilePath,
+	}, nil
+}
+
+// Parse parses the raw gosec data into a known type.
+func (p *parser) Parse() (*sarif.Report, error) {
+	b, err := os.ReadFile(p.rawOutFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file '%s': %w", p.rawOutFilePath, err)
+	}
+
+	// TODO - BEGIN
+	// This will be replaced by https://linear.app/smithy/issue/OCU-277/support-openapi-spec-for-sarif.
```

Comment:
nit: wanna replace this with the equivalent github issue for visibility? this is a public repo and not eveyone has access to linear